test/processStates_test.js

```diff
@@ -0,0 +1,260 @@
+goog.module('datalayerhelper.helper.testing.processStates');
+goog.setTestOnly();
+
+const {DataLayerHelper, buildAbstractModelInterface_} = goog.require('helper');
+
+describe('The processStates function', () => {
+  /**
+   * Asserts that calling processStates_ with the given arguments will result
+   * in the expected model state and expected calls to the listener.
+   *
+   * @param {!Array<Object>} states The states argument for the call to
+   *     processStates_.
+   * @param {!Object<*>} expectedModel The expected model
+   *     state after the call.
+   * @param {!Array<Array<*>>} expectedListenerCalls The expected calls made
+   *     to the helper's listener. Should be an array of arrays where each sub-
+   *     array represents the arguments passed to the listener during a call.
+   */
+  function assertProcessStates(states, expectedModel, expectedListenerCalls) {
+    const MockHelper = function() {
```

Comment:
arrow function here and everywhere below?

Reply:
All of the functions in the class that are not arrow functions have to be normal functions, either because they use special values of this or because they use the arguments variable. I think that this section of code is particularly outdated/confusing though, so I simplified it and got rid of this function/class entirely.